logstash-core/lib/logstash/pipeline.rb

```diff
@@ -447,12 +447,14 @@ def worker_loop(batch_size, batch_delay)
       shutdown_requested |= signal.shutdown? # latch on shutdown signal
 
       batch = @filter_queue_client.read_batch # metrics are started in read_batch
```

Comment:
I could probably answer that in about 3 days 😄 , but hopefully someone more familiar can chime in.